src/test/java/org/openrewrite/maven/InitMojoIT.java

```diff
@@ -0,0 +1,32 @@
+package org.openrewrite.maven;
+
+import com.soebes.itf.jupiter.extension.MavenGoal;
+import com.soebes.itf.jupiter.extension.MavenJupiterExtension;
+import com.soebes.itf.jupiter.extension.MavenTest;
+import com.soebes.itf.jupiter.extension.SystemProperties;
+import com.soebes.itf.jupiter.extension.SystemProperty;
+import com.soebes.itf.jupiter.maven.MavenExecutionResult;
+
+import static com.soebes.itf.extension.assertj.MavenITAssertions.assertThat;
+
+@MavenJupiterExtension
+@MavenGoal("${project.groupId}:${project.artifactId}:${project.version}:init")
+@SystemProperties({
+        @SystemProperty(value = "activeRecipes", content = "org.openrewrite.java.testing.junit5.ParameterizedRunnerToParameterized"),
+        @SystemProperty(value = "dependencies", content = "<dependencies><dependency><groupId>org.openrewrite.recipe</groupId><artifactId>rewrite-spring</artifactId><version>4.14.1</version></dependency></dependencies>"),
+        @SystemProperty(value = "rootOnly", content = "false")
```

Comment:
There seems to be a mismatch in the first conditional in InitMojo, which is why this rootOnly=false is needed; didn't know if there was another proper fix, but this works, and InitMojo works in practice as well.